smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/PackageManifestGenerator.kt

```diff
@@ -4,111 +4,72 @@
-import software.amazon.smithy.utils.CodeWriter
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
 
-fun writePackageManifest(settings: SwiftSettings, fileManifest: FileManifest, dependencies: List<SymbolDependency>, generateTestTarget: Boolean = false) {
+class PackageManifestGenerator(val ctx: ProtocolGenerator.GenerationContext) {
 
-    // filter duplicates in dependencies
-    val distinctDependencies = dependencies.distinctBy { it.packageName }
-    val writer = CodeWriter().apply {
-        trimBlankLines()
-        trimTrailingSpaces()
-        setIndentText("    ")
-    }
+    fun writePackageManifest(dependencies: List<SymbolDependency>) {
+        ctx.delegator.useFileWriter("Package.swift") { writer ->
+            writer.write("// swift-tools-version:\$L", ctx.settings.swiftVersion)
+            writer.write("")
+            writer.write("import PackageDescription")
+            writer.write("")
 
-    writer.write("// swift-tools-version:${settings.swiftVersion}")
-    writer.write("")
-    writer.write("import PackageDescription")
-    writer.write("import class Foundation.ProcessInfo")
-    writer.write("import class Foundation.FileManager")
+            writer.openBlock("let package = Package(", ")") {
+                writer.write("name: \$S,", ctx.settings.moduleName)
 
-    writer.openBlock("let package = Package(", ")") {
-        writer.write("name: \"${settings.moduleName}\",")
+                writer.openBlock("platforms: [", "],") {
+                    writer.write(".macOS(.v10_15), .iOS(.v13)")
+                }
 
-        writer.openBlock("platforms: [", "],") {
-            writer.write(".macOS(.v10_15), .iOS(.v13)")
-        }
+                writer.openBlock("products: [", "],") {
+                    writer.write(".library(name: \$S, targets: [\$S])", ctx.settings.moduleName, ctx.settings.moduleName)
+                }
 
-        writer.openBlock("products: [", "],") {
-            writer.write(".library(name: \"${settings.moduleName}\", targets: [\"${settings.moduleName}\"])")
-        }
+                val externalDependencies = dependencies.filter { it.getProperty("url", String::class.java).get().isNotEmpty() }
+                val dependenciesByURL = externalDependencies.distinctBy { it.expectProperty("url", String::class.java) }
 
-        writer.openBlock("targets: [", "]") {
-            writer.openBlock(".target(", "),") {
-                writer.write("name: \"${settings.moduleName}\",")
                 writer.openBlock("dependencies: [", "],") {
-                    for (dependency in distinctDependencies) {
-                        writer.openBlock(".product(", "),") {
-                            val target = dependency.expectProperty("target", String::class.java)
-                            writer.write("name: \"${target}\",")
-                            writer.write("package: \"${dependency.packageName}\"")
+                    dependenciesByURL.forEach { dependency ->
```

Comment:
All the old logic for reading env vars in Package.swift is removed.

For now, Package.swift will get dependencies from local folders... in the future, it will point to published package versions.